subprojects/emcee/android-device/src/main/kotlin/com/avito/android/device/internal/AndroidDeviceImpl.kt

```diff
@@ -16,29 +15,24 @@ import com.malinskiy.adam.request.testrunner.TestRunnerRequest
 import kotlinx.coroutines.ExperimentalCoroutinesApi
 import kotlinx.coroutines.channels.receiveOrNull
 import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.runBlocking
+import java.util.logging.Logger
 
 internal class AndroidDeviceImpl(
     override val sdk: Int,
     override val type: String,
     override val serial: DeviceSerial,
+    private val installPackage: InstallPackage,
     private val adb: AndroidDebugBridgeClient,
 ) : AndroidDevice {
 
+    private val logger = Logger.getLogger("AndroidDevice")
+
     override suspend fun install(application: AndroidApplication): Result<Unit> {
         return execute(
             action = {
-                val success = adb.execute(
-                    StreamingPackageInstallRequest(
-                        pkg = application.apk,
-                        supportedFeatures = emptyList(),
-                        reinstall = false,
-                        extraArgs = emptyList()
-                    ),
-                    serial = serial.value
-                )
-                if (!success) {
-                    throw RuntimeException("Streaming install of the $application not success")
-                }
+                logger.info("Start install $application")
+                runBlocking { installPackage.installTo(application, serial) }
```

Comment:
What is install package here? Why do we pass application in `installTo` method signature? Class and method name are unclear to me.